packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -146,56 +144,117 @@ function ManualGridVisualizer( { gridClientId, gridInfo } ) {
-			return (
-				<GridVisualizerCell
-					key={ `${ row }-${ column }` }
-					color={ gridInfo.currentColor }
-					className={ isHighlighted && 'is-highlighted' }
-				>
-					{ isCellOccupied ? (
-						<GridVisualizerDropZone
-							column={ column }
-							row={ row }
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-							setHighlightedRect={ setHighlightedRect }
-						/>
-					) : (
-						<GridVisualizerAppender
-							column={ column }
-							row={ row }
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-							setHighlightedRect={ setHighlightedRect }
-						/>
-					) }
-				</GridVisualizerCell>
-			);
-		} )
+	return (
+		<BlockPopoverCover
+			__unstablePopoverSlot="__unstable-block-tools-after"
+			className={ clsx( 'block-editor-grid-visualizer', {
+				'is-dropping-allowed': isDroppingAllowed,
+			} ) }
+			clientId={ gridClientId }
+		>
+			<div
+				ref={ ref }
+				className="block-editor-grid-visualizer__grid"
+				style={ gridInfo.style }
+			>
+				{ range( 1, gridInfo.numRows ).map( ( row ) =>
+					range( 1, gridInfo.numColumns ).map( ( column ) => {
+						const isCellOccupied = occupiedRects.some( ( rect ) =>
+							rect.contains( column, row )
+						);
+						const isHighlighted =
+							highlightedRect?.contains( column, row ) ?? false;
+
+						return (
+							<div
+								key={ `${ row }-${ column }` }
+								className={ clsx(
+									'block-editor-grid-visualizer__cell',
+									{
+										'is-highlighted': isHighlighted,
+									}
+								) }
+							>
+								{ isCellOccupied ? (
+									<GridVisualizerDropZone
+										column={ column }
+										row={ row }
+										gridClientId={ gridClientId }
+										gridInfo={ gridInfo }
+										setHighlightedRect={
+											setHighlightedRect
+										}
+									/>
+								) : (
+									<GridVisualizerAppender
+										column={ column }
+										row={ row }
+										gridClientId={ gridClientId }
+										gridInfo={ gridInfo }
+										setHighlightedRect={
+											setHighlightedRect
+										}
+									/>
+								) }
+							</div>
+						);
+					} )
+				) }
+			</div>
+		</BlockPopoverCover>
 	);
-}
+} );
 
-function GridVisualizerCell( { color, children, className } ) {
-	return (
-		<div
-			className={ clsx(
-				'block-editor-grid-visualizer__cell',
-				className
-			) }
-			style={ {
-				boxShadow: `inset 0 0 0 1px color-mix(in srgb, ${ color } 20%, #0000)`,
+/**
+ * A popover component that renders inline under the grid block.
+ *
+ * This provides non-interactive elements of the grid visualization and
+ * renders under the block so that the background colors are not atop
+ * the block content.
+ *
+ * @param {Object}  props
+ * @param {string}  props.gridClientId
+ * @param {Object}  props.gridInfo
+ * @param {boolean} props.isManualGrid
+ */
+function GridPopunder( { gridClientId, gridInfo, isManualGrid } ) {
```

Comment:
`GridPopunder` (not sure if the name is too twee) now renders the blue cells in the background. It's pretty similar to the auto mode cells in `trunk` in terms of code.